ldap/servers/slapd/back-ldbm/db-bdb/bdb_layer.c

```diff
@@ -6948,7 +6949,13 @@ bdb_public_private_open(backend *be, const char *db_filename, int rw, dbi_env_t
         conf->bdb_tx_max = 50;
         rc = bdb_start(li, DBLAYER_NORMAL_MODE);
         if (rc == 0) {
-            bdb_env = ((struct bdb_db_env*)(priv->dblayer_env))->bdb_DB_ENV;
+            pEnv = (bdb_db_env *)priv->dblayer_env;
+            if (pEnv == NULL) {
+                fprintf(stderr, "bdb_public_private_open: dbenv is not available (0x%p) for database %s\n",
+                        (void *)pEnv, db_filename ? db_filename : "unknown");
+                return EINVAL;
```

Comment:
Okay, after some "read bdb flow" time I checked and found the issue.
Actually, it was a mutex issue in bdb_public_private_close because at some point we might have already closed the dblayer env. Anyway, hopefully it's fixed now.